heartbeat/monitors/monitor.go

```diff
@@ -71,6 +72,12 @@ type Monitor struct {
 	stats plugin.RegistryRecorder
 
 	monitorStateTracker *monitorstate.Tracker
+	statusReporter      status.StatusReporter
+}
+
+// SetStatusReporter
+func (m *Monitor) SetStatusReporter(statusReporter status.StatusReporter) {
```

Comment:
Since its set on the Monitor level, what happens if multiple monitors were configured, does the errors get accumulated or there is upper limit to how its shown in the UI ?

Reply:
Every monitor will map 1:1 to an agent integration, which Fleet UI already shows individually: